test/worker_test.py

```diff
@@ -1264,3 +1265,50 @@ def complete(self):
                           ['KeyboardInterruptTask', '--local-scheduler', '--no-lock'])
         self.assertRaises(KeyboardInterrupt, luigi_run,
                           ['ExternalKeyboardInterruptTask', '--local-scheduler', '--no-lock'])
+
+
+class WorkerPurgeEventHandlerTest(unittest.TestCase):
+
+    def test_process_killed_handler(self):
+        result = []
+
+        @HungWorker.event_handler(Event.PROCESS_FAILURE)
+        def store_task(t, error_msg):
+            result.append(t)
+
+        w = Worker(worker_processes=2, wait_interval=0.01, timeout=5)
+        task = HungWorker()
+        w.add(task)
+        w._run_task(task.task_id)
+
+        processes = w._running_tasks.values()
+        self.assertEqual(1, len(processes))
+        process = processes[0]
+
+        # kill and wait for it to be killed.
+        os.kill(process.pid, signal.SIGUSR2)
+        while process.is_alive():
+            time.sleep(.1)
```

Comment:
Couldn't you just create a mock TaskProcess that returns False for is_alive and has a nonzero exit code?